pallets/market/src/lib.rs

```diff
@@ -300,6 +300,47 @@ pub mod pallet {
         pub client_signature: OffchainSignature,
     }
 
+    /// Bounds for deal duration that storage providers want to accept.
+    /// Used in the [`DealParameters`]
+    #[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+    pub struct DealDurationBound<BlockNumber> {
+        pub lower: Option<BlockNumber>,
+        pub upper: Option<BlockNumber>,
+    }
+
+    /// Deal Parameters set by storage providers on which deals they accept,
+    /// based on deal variables.
+    #[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+    pub struct DealParameters<Balance, BlockNumber> {
+        pub minimum_price_per_block: Balance,
+        pub deal_duration: DealDurationBound<BlockNumber>,
+    }
+
+    impl<Balance, BlockNumber> DealParameters<Balance, BlockNumber>
+    where
+        Balance: PartialOrd,            // For comparison
+        BlockNumber: PartialOrd + Copy, // Copy so we can match the Option
```

Comment:
Instead of the comments you left, I'd write something like:
```suggestion
    // `Balance` and `BlockNumber` are not directly related to their `Config` counterparts,
    // the structure is generic enough that you could use it for other purposes,
    // hence we just restrict the generics to the strictly necessary subset of traits.
    //
    // Additional note: `BlockNumber` will usually be a number, especially in our network, 
    // as such, the `Copy` trait should come for free and not require any extra work on 
    // further implementations; furthermore, the actual `BlockNumber` does require `Copy`
    // https://docs.rs/sp-runtime/40.1.0/sp_runtime/traits/trait.BlockNumber.html
    impl<Balance, BlockNumber> DealParameters<Balance, BlockNumber>
    where
        Balance: PartialOrd,           
        BlockNumber: PartialOrd + Copy,
```